dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/VisitorLitmusC.java

```diff
@@ -28,23 +30,24 @@
 
 public class VisitorLitmusC extends LitmusCBaseVisitor<Object> {
 
-    private final ExpressionFactory expressions = ExpressionFactory.getInstance();
-    private final ProgramBuilder programBuilder;
-    private final IntegerType archType = TypeFactory.getInstance().getArchType();
+    private final ProgramBuilder programBuilder = ProgramBuilder.forLanguage(Program.SourceLanguage.LITMUS);
+    private final ExpressionFactory expressions = programBuilder.getExpressionFactory();
+    private final IntegerType archType = programBuilder.getTypeFactory().getArchType();
     private int currentThread;
     private int scope;
     private int ifId = 0;
     private Register returnRegister;
 
-    public VisitorLitmusC(ProgramBuilder pb){
-        this.programBuilder = pb;
+    public VisitorLitmusC(){
     }
 
     // ----------------------------------------------------------------------------------------------------------------
     // Entry point
 
     @Override
     public Program visitMain(LitmusCParser.MainContext ctx) {
+        //FIXME: We should visit thread declarations before variable declarations
+        // because variable declaration refer to threads.
```

Comment:
Why is this not causing problems right now? It is because of line 85?

Also, don't we have the same issue with other litmus parsers?

Reply:
The other Litmus parser work properly by first visiting all thread declarations.
I guess the LitmusC parser doesn't do this, because its C-like syntax is different to the usual Litmus syntax?
Anyway, the functions of the LitmusC parser are written so that they create a thread if they encounter a variable/register that refers to that thread.
I didn't have the nerves to also fix the LitmusC parser in this PR, so I just put comments on all its weird behaviors :).